src/Select/index.js

```diff
@@ -289,25 +293,62 @@ const Select = ({
         >
           {showMenu &&
             hasCategories &&
-            categories.map(({ label, categoryChildren }) => (
-              <details
-                key={label}
-                className="nds-select-category"
-                {...getDetailsProps(categoryChildren)} // controls open state
-              >
-                <summary className="fontWeight--bold alignChild--left--center padding--x--s padding--y-xs">
-                  <span id={`select-category-${label}`}>{label}</span>
-                  <span className="nds-category-icon narmi-icon-chevron-down" />
-                  <span className="nds-category-icon narmi-icon-chevron-up" />
-                </summary>
-                <ul
-                  className="list--reset"
-                  aria-labelledby={`select-category-${label}`}
+            categories.map(({ label, kind, categoryChildren, isFlat }) => {
+              return isFlat ? (
```

Comment:
`isFlat ?` render a set of flat lists `:` render as [summary/details](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details)